### PR TITLE
Disable blob deduping temporarily

### PR DESCRIPTION
**Pull request by jatgarg**
We can enable slowly in future. Allowing host to set policy so that it can be turned on without making changes at driver layer.
Currently we are facing Data corruption issue where directory blob is wrongly showing as map blob. We already disabled handle expansion but for now to be safe we are disabling entire blob deduping. Currently we are not facing the issue after disabling handle expansion but we can monitor telemetry before making any further conclusions.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts [hunk 1]
```diff
@@ -369,15 +374,20 @@ export class OdspSummaryUploadManager {
...
+                            hash = await hashFile(IsoBuffer.from(value.content, value.encoding));
+                            cachedPath = this.blobTreeDedupCaches.blobShaToPath.get(hash);
+                        }
                     }
                     (summaryObject as any).content = undefined;
                     // If the cache has the hash of the blob and handle of last summary is also present, then use that
                     // cached path for the given blob. Also update the caches for future use.
                     if (cachedPath === undefined || parentHandle === undefined) {
                         blobs++;
                     } else {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts:387]: "else if" on one line

File: packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts [hunk 2]
```diff
@@ -369,15 +374,20 @@ export class OdspSummaryUploadManager {
...
+                        if (!this.hostPolicy.blobDeduping) {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts:388]: Not sure here. If this is the case we should never get to, then it should be an assert. If it's normal path to get to, then it should not be an error event

File: packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts [hunk 3]
```diff
@@ -416,7 +426,7 @@ export class OdspSummaryUploadManager {
                         reusedBlobs += result.reusedBlobs;
                         blobs += result.blobs;
                     } else {
-                        if (allowHandleExpansion) {
+                        if (allowHandleExpansion && this.hostPolicy.blobDeduping) {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts:429]: It's a bit verbose with this.hostPolicy.blobDeduping, considert passing boolean to this function / class or creating variable in function with it.

File: packages/drivers/odsp-driver/src/test/odspSummaryUploadManagerTests.spec.ts [hunk 4]
```diff
@@ -31,6 +31,7 @@ describe("Odsp Summary Upload Manager Tests", () => {
             async (options: TokenFetchOptions, name?: string) => "token",
             logger,
             epochTracker,
+            { blobDeduping: true },
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/test/odspSummaryUploadManagerTests.spec.ts:34]: Would be great to test with and without (i.e. shipping code as well :))

**Review by vladsud — APPROVED** [review]

**Comment by anthony-murphy** [comment]
My only concern here is around whether we will ever turn this back on. If there isn't a plan to turn it back on i'd rather just remove the code, and unused code will just accur dept

File: packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts (hunk 2 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts:388]: Added assert. Should never come here when blob deduping is disabled.

File: packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts (hunk 1 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts:387]: If is a part of else here.

File: packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts (hunk 3 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts]: Added var.

File: packages/drivers/odsp-driver/src/test/odspSummaryUploadManagerTests.spec.ts (hunk 4 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/test/odspSummaryUploadManagerTests.spec.ts:34]: added

**Comment by jatgarg** [comment]
> My only concern here is around whether we will ever turn this back on. If there isn't a plan to turn it back on i'd rather just remove the code, as unused code will just accur dept

Added an issue for this: https://github.com/microsoft/FluidFramework/issues/5444

**Pull request merged**